### PR TITLE
feat(jtk): fix fields list column order, rename --custom flag, and add fields show

### LINKED ISSUE #285: jtk fields: fix list column order, rename --custom flag, and add fields show
Parent: #230

### `jtk fields list` (default)
**Current:** extra CUSTOM column; NAME/TYPE column order swapped vs spec.
**Expected:**
```
ID | TYPE | NAME
summary | string | Summary
status | status | Status
customfield_10035 | number | Story Points
customfield_10050 | option | Team
customfield_10020 | array | Sprint
```

### `jtk fields list --id`
**Current:** flag not wired — output identical to default.
**Expected:** one field ID per line.

### `jtk fields list --custom` (spec: `--custom-fields`)
**Current:** flag is `--custom`; spec says `--custom-fields` (to match `jtk issues fields --custom-fields`).
**Action:** rename flag to `--custom-fields` (keep `--custom` as a hidden alias if needed).

### `jtk fields list --extended`
**Current:** extra CUSTOM column; column ordering differs.
**Expected:**
```
ID | TYPE | SEARCHABLE | NAVIGABLE | ORDERABLE | CLAUSE_NAMES | NAME
summary | string | yes | yes | yes | summary | Summary
status | status | yes | yes | no | status | Status
customfield_10035 | number | yes | yes | yes | cf[10035], Story Points | Story Points
customfield_10050 | option | yes | yes | yes | cf[10050], Team[Dropdown], Team | Team
```

### `jtk fields show <fieldId>`
**Current:** command missing — `fields` only has `list`, `contexts`, `options`.
**Expected:** flat denormalized context/option view:
```
CONTEXT_ID | CONTEXT | PROJECTS | OPTION_ID | OPTION_VALUE
10100 | Default Context | (global) | 20001 | Platform
10100 | Default Context | (global) | 20002 | Integration
10101 | MON Project Context | MON | 20010 | CapOne
10102 | ON Project Context | ON | - | -
```
Empty contexts render with `- | -` so the context is still discoverable.

See #230 for full spec.

### PR DESCRIPTION
**Pull request by rianjs**
## Summary
- Fix `fields list` default column order to `ID|TYPE|NAME` (removes extra CUSTOM column)
- Fix `fields list --extended` to `ID|TYPE|SEARCHABLE|NAVIGABLE|ORDERABLE|CLAUSE_NAMES|NAME`
- Wire `--id` flag for `fields list` (was not wired)
- Rename `--custom` to `--custom-fields` (hidden `--custom` alias for backwards compat)
- Add `fields show <field-id>` — flat denormalized context/option view: `CONTEXT_ID|CONTEXT|PROJECTS|OPTION_ID|OPTION_VALUE`
- Add paginated API methods: `GetAllFieldContexts`, `GetAllFieldContextProjectMappings`, `GetAllFieldContextOptions`

Closes #285

## Test plan
- [x] Presenter tests: PresentList default/extended column order, PresentFieldShow
- [x] Command tests: --id, --custom-fields, --custom hidden alias, column order assertions
- [x] Show command: table, JSON, --id, empty contexts, 4xx error tolerance for non-select fields
- [x] `make build` / `make test` / `make lint` pass

### CODE REVIEW

**Comment by rianjs** [comment]
**Major**
- `tools/jtk/internal/cmd/fields/show.go:47` handles empty contexts before output-mode precedence. `jtk fields show X --id` will emit `No contexts found...` instead of empty ID output, and `jtk fields show X -o json` will emit non-JSON text instead of `[]`. Move the empty handling after `--id` / JSON branching, or build `[]FieldShowRow{}` first and render by mode.

**Minor**
- `tools/jtk/api/field_management.go:279` adds the core page-all methods, but the shown tests only cover single-page `isLast: true` responses. Add multi-page tests for contexts, project mappings, and options so the #285 pagination contract is locked.
- `tools/jtk/internal/cmd/fields/fields_test.go:1271` only proves 400 is swallowed. Add at least one 500 or 429 option-fetch test to prove non-tolerable errors still propagate.

Test note: I attempted `go test ./api ./internal/present ./internal/cmd/fields`, but this sandbox cannot bind `httptest` ports, so I could not independently verify the suite.

**Comment by rianjs** [comment]
**Blocker**
- None.

**Major**
- None.

**Minor**
- Page-all test coverage is still incomplete. `TestRunShow_MultiPageContexts` exercises contexts only, and it does not assert the requested `startAt` values. Since page-all behavior for contexts, project mappings, and options is part of the architectural contract, add focused tests for all three exported methods that verify `startAt` progression and combined returned values.

**Nit**
- None.

Test note: I did not rerun the suite in this sandbox; `httptest` port binding is blocked here as before.

**Comment by rianjs** [comment]
## TDD Assessment

### What's Well-Covered

**Pagination (all three new methods)** — `TestGetAllFieldContexts_MultiPage`, `TestGetAllFieldContextProjectMappings_MultiPage`, and `TestGetAllFieldContextOptions_MultiPage` all exercise the pagination loop with a two-page scenario: first page returns `isLast: false`, second returns `isLast: true`, and call count is asserted. That's the core contract for these helpers.

**`fields list` column order** — `TestRunList_ColumnOrder` verifies the default (ID|TYPE|NAME) data-row cell positions directly. `TestRunList_Extended_ColumnOrder` verifies all 7 cells in order (ID|TYPE|SEARCHABLE|NAVIGABLE|ORDERABLE|CLAUSE_NAMES|NAME). Both are backed by `TestFieldPresenter_PresentList_Default` and `TestFieldPresenter_PresentList_Extended` at the presenter layer.

**`--id` flag wiring for `fields list`** — `TestRunList_IDOnly` confirms one-per-line output; `TestRunList_IDOnly_Empty` confirms silent success on no results. Both exercise `runList` directly with `IDOnly: true` set on opts.

**`--custom-fields` rename and hidden `--custom` alias** — `TestNewListCmd` checks that `custom-fields` flag exists with default `"false"` and that `custom` flag exists and is hidden. Both point to the same `customOnly` variable (two `BoolVar` calls to the same destination — valid in cobra).

**`fields show` happy path** — `TestRunShow_Table` covers the full denormalization: global context maps to `(global)`, project context maps to project IDs, context with no options renders `- | -`. Header presence for `CONTEXT_ID` and `OPTION_VALUE` is checked. JSON and ID-only variants also tested.

**Error tolerance** — `TestRunShow_OptionFetchError4xx` (HTTP 400) confirms the context row is still emitted with `-` sentinels and no error is returned. `TestRunShow_OptionFetchError5xx` (HTTP 500) confirms the error propagates up and contains the server error message.

**`fields show` empty states** — `TestRunShow_Empty` (table), `TestRunShow_EmptyJSON` (JSON `[]`), and `TestRunShow_EmptyIDOnly` (no output) all gate on zero contexts, which is the early-exit path before project mappings are fetched.

---

### Gaps and Things Worth Flagging

**`--custom-fields` flag functional test is missing.** `TestNewListCmd` verifies the flag exists, but there is no test that passes `--custom-fields` through cobra's `Execute` path and confirms filtering fires. All existing custom-filter tests call `runList(ctx, opts, true, "")` directly, bypassing flag parsing entirely. If someone fat-fingers the `BoolVar` destination binding, no test would catch it at the integration boundary.

**`isOptionFetchTolerable` only covers `ErrBadRequest` (400), not `ErrNotFound` (404).** The function tolerates both `ErrBadRequest` and `ErrNotFound`, but only a 400 server is exercised in `TestRunShow_OptionFetchError4xx`. There is no test that serves HTTP 404 and confirms it is swallowed. `ErrForbidden` (403) is notably absent from `isOptionFetchTolerable` — Jira returns 403 on option fetches for non-select fields in some contexts — worth deciding whether 403 should be tolerated and, if so, adding a test.

**`buildProjectMap` context-is-both-global-and-project-mapped edge case is untested.** The function has a guard (`if _, isGlobal := result[ctxID]; isGlobal { continue }`) that prevents a project mapping from overwriting a global designation, but no test exercises a response where the same context ID appears in both global and project mapping lists. This is probably an API invariant, but it's an untested branch.

**`fields show --id` deduplication logic is dead code in practice.** The dedup map (`seen`) in `runShow` exists because `GetAllFieldContexts` could theoretically return the same context ID twice across pages. In practice `GetAllFieldContexts` paginates by offset and the Jira API won't duplicate context IDs, so this is defensive code with no test that actually triggers it.

**`TestRunShow_Table` header assertion is incomplete.** Only `CONTEXT_ID` and `OPTION_VALUE` are checked in the header (the first and last columns). `CONTEXT`, `PROJECTS`, and `OPTION_ID` are not asserted. If a column were reordered or renamed, the test might not catch it. `TestFieldPresenter_PresentFieldShow` at the presenter layer does assert all five headers in order, so this is only a gap in the integration-level test.

**Pagination terminal condition: empty `Values` slice.** Both `GetAllFieldContexts` and `GetAllFieldContextProjectMappings` break on `page.IsLast || len(page.Values) == 0`. The `len(page.Values) == 0` branch (server returns non-last page with zero items — a malformed response) has no test. Low priority, but worth noting for completeness.

**No test for `fields show` with `--id` and a single-page but multi-context response where some contexts are duplicated across pages.** The ID-only dedup path (`seen` map) is never actually exercised.

---

### Summary

The critical paths — pagination correctness, column order, `--id` emission, 4xx error tolerance, 5xx propagation, and all three empty-state variants — are covered. The main actionable gaps are: (1) a cobra-level functional test for `--custom-fields` to close the flag-wiring risk, and (2) a 404 server test for `isOptionFetchTolerable` to match the declared contract. Everything else is low-risk defensive code.

File: tools/jtk/api/field_management.go [hunk 1]
```diff
@@ -260,6 +260,101 @@ func (c *Client) UpdateFieldContextOptions(ctx context.Context, fieldID, context
...
+// GetAllFieldContextOptions returns all options for a field context, paginating through all pages.
+func (c *Client) GetAllFieldContextOptions(ctx context.Context, fieldID, contextID string) ([]FieldContextOption, error) {
+	if fieldID == "" {
+		return nil, ErrFieldIDRequired
+	}
+	var all []FieldContextOption
+	startAt := 0
+	for {
+		urlStr := fmt.Sprintf("%s/field/%s/context/%s/option?startAt=%d",
+			c.BaseURL, url.PathEscape(fieldID), url.PathEscape(contextID), startAt)
```
Comment (monit-reviewer) [inline tools/jtk/api/field_management.go:340]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): GetAllFieldContextOptions returns raw errors from c.Get and json.Unmarshal without wrapping context (e.g. `fmt.Errorf("fetching field context options: %w", err)`), unlike sibling methods GetAllFieldContexts and GetAllFieldContextProjectMappings which both wrap errors for diagnostic context. The inconsistency makes it harder to identify the call site from logs or stack traces.

*Reply to this thread when addressed.*

File: tools/jtk/api/field_management.go [hunk 2]
```diff
@@ -260,6 +260,101 @@ func (c *Client) UpdateFieldContextOptions(ctx context.Context, fieldID, context
...
+		}
+		all = append(all, page.Values...)
+		if page.IsLast || len(page.Values) == 0 {
+			break
+		}
+		startAt += len(page.Values)
+	}
+	return all, nil
+}
+
```
Comment (monit-reviewer) [inline tools/jtk/api/field_management.go:330]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): GetAllFieldContextOptions validates fieldID but not contextID. An empty contextID produces a malformed URL path `/field/{id}/context//option`. While contextID always comes from the API in practice, the missing guard is inconsistent with the fieldID validation in this method and breaks the defensive pattern of its siblings.

*Reply to this thread when addressed.*

File: tools/jtk/api/field_management_test.go [hunk 3]
```diff
@@ -359,3 +359,97 @@ func TestDeleteFieldContextOption_EmptyID(t *testing.T) {
 	err := client.DeleteFieldContextOption(context.Background(), "", "10001", "3")
```
Comment (monit-reviewer) [inline tools/jtk/api/field_management_test.go:359]: 🔵 **Low** (harness-engineering:harness-enforcement-reviewer): No test covers GetAllFieldContextOptions called with an empty contextID. Given the missing validation guard in the implementation, a test like TestGetAllFieldContextOptions_EmptyContextID would either catch the missing guard or document that empty contextID is intentionally permitted.

*Reply to this thread when addressed.*

File: tools/jtk/api/field_management.go [hunk 4]
```diff
@@ -260,6 +260,101 @@ func (c *Client) UpdateFieldContextOptions(ctx context.Context, fieldID, context
...
+// GetAllFieldContexts returns all contexts for a field, paginating through all pages.
+func (c *Client) GetAllFieldContexts(ctx context.Context, fieldID string) ([]FieldContext, error) {
+	if fieldID == "" {
+		return nil, ErrFieldIDRequired
+	}
+	var all []FieldContext
+	startAt := 0
+	for {
+		urlStr := fmt.Sprintf("%s/field/%s/context?startAt=%d", c.BaseURL, url.PathEscape(fieldID), startAt)
+		body, err := c.Get(ctx, urlStr)
```
Comment (monit-reviewer) [inline tools/jtk/api/field_management.go:288]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): None of the three new paginated methods pass a `maxResults` query parameter, so the API uses its default (typically 50). For fields with many contexts or options this increases round-trips unnecessarily. Consider adding `&maxResults=100` (or the API-documented maximum) to each URL.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/fields/show.go [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+				ContextID:   fc.ID,
+				Context:     fc.Name,
+				Projects:    projects,
+				OptionID:    opt.ID,
+				OptionValue: opt.Value,
+			})
+		}
+	}
+	return rows, nil
+}
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/fields/show.go:159]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): isOptionFetchTolerable silently swallows 404 in addition to 400. A 404 from GET /field/{id}/context/{ctxId}/option can mean the context was deleted between the contexts fetch and the options fetch — a real data-consistency issue — not only that the field is a non-select type. Only 400 ('field is not a select/option type') is semantically safe to swallow here.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/fields/show.go [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+		if _, isGlobal := result[ctxID]; isGlobal {
+			continue
+		}
+		sort.Strings(ids)
+		result[ctxID] = strings.Join(ids, ", ")
+	}
+	return result
+}
+
+func buildShowRows(ctx context.Context, client *api.Client, fieldID string, contexts []api.FieldContext, projectsByContext map[string]string) ([]jtkpresent.FieldShowRow, error) {
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/fields/show.go:114]: 🔵 **Low** (harness-engineering:harness-self-documenting-code-reviewer): isOptionFetchTolerable suppresses 400/404 errors and emits a '-'/'-' row with no stderr notice. A user running `jtk fields show` on a text field will see context rows with no options and no indication of why, which is indistinguishable from a field that legitimately has no options.

*Reply to this thread when addressed.*

File: tools/jtk/internal/cmd/fields/fields_test.go [hunk 7]
```diff
@@ -1005,3 +1149,380 @@ func TestRunList_CacheHit_CustomFilter_SkipsLiveCall(t *testing.T) {
...
+	t.Parallel()
+	server := newShowTestServer(t)
+	defer server.Close()
+
+	client, err := api.New(api.ClientConfig{URL: server.URL, Email: "test@test.com", APIToken: "token"})
+	testutil.RequireNoError(t, err)
+
+	var stdout bytes.Buffer
+	opts := &root.Options{Output: "table", Stdout: &stdout, Stderr: &bytes.Buffer{}}
+	opts.SetAPIClient(client)
```
Comment (monit-reviewer) [inline tools/jtk/internal/cmd/fields/fields_test.go:1213]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): TestRunShow_Table asserts the 10102 (empty) row by hardcoding `lines[4]`, assuming exactly 3 data rows precede it (2 options for 10100, 1 for 10101). If server response ordering or row spacing changes, the assertion silently tests the wrong row. The test should search for the 10102 row by content rather than absolute position.

*Reply to this thread when addressed.*

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `efaf501`

> **Approved** with 7 non-blocking suggestions below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-architecture-reviewer | 4 |
| harness-engineering:harness-enforcement-reviewer | 1 |
| harness-engineering:harness-self-documenting-code-reviewer | 2 |

<details>
<summary><strong>harness-engineering:harness-architecture-reviewer (4 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/api/field_management.go:340`
> GetAllFieldContextOptions returns raw errors from c.Get and json.Unmarshal without wrapping context (e.g. `fmt.Errorf("fetching field context options: %w", err)`), unlike sibling methods GetAllFieldContexts and GetAllFieldContextProjectMappings which both wrap errors for diagnostic context. The inconsistency makes it harder to identify the call site from logs or stack traces.

**💡 Suggestion** - `tools/jtk/api/field_management.go:330`
> GetAllFieldContextOptions validates fieldID but not contextID. An empty contextID produces a malformed URL path `/field/{id}/context//option`. While contextID always comes from the API in practice, the missing guard is inconsistent with the fieldID validation in this method and breaks the defensive pattern of its siblings.

**💡 Suggestion** - `tools/jtk/internal/cmd/fields/show.go:159`
> isOptionFetchTolerable silently swallows 404 in addition to 400. A 404 from GET /field/{id}/context/{ctxId}/option can mean the context was deleted between the contexts fetch and the options fetch — a real data-consistency issue — not only that the field is a non-select type. Only 400 ('field is not a select/option type') is semantically safe to swallow here.

**💡 Suggestion** - `tools/jtk/internal/cmd/fields/fields_test.go:1213`
> TestRunShow_Table asserts the 10102 (empty) row by hardcoding `lines[4]`, assuming exactly 3 data rows precede it (2 options for 10100, 1 for 10101). If server response ordering or row spacing changes, the assertion silently tests the wrong row. The test should search for the 10102 row by content rather than absolute position.

</details>

<details>
<summary><strong>harness-engineering:harness-enforcement-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/api/field_management_test.go:359`
> No test covers GetAllFieldContextOptions called with an empty contextID. Given the missing validation guard in the implementation, a test like TestGetAllFieldContextOptions_EmptyContextID would either catch the missing guard or document that empty contextID is intentionally permitted.

</details>

<details>
<summary><strong>harness-engineering:harness-self-documenting-code-reviewer (2 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/api/field_management.go:288`
> None of the three new paginated methods pass a `maxResults` query parameter, so the API uses its default (typically 50). For fields with many contexts or options this increases round-trips unnecessarily. Consider adding `&maxResults=100` (or the API-documented maximum) to each URL.

**💡 Suggestion** - `tools/jtk/internal/cmd/fields/show.go:114`
> isOptionFetchTolerable suppresses 400/404 errors and emits a '-'/'-' row with no stderr notice. A user running `jtk fields show` on a text field will see context rows with no options and no indication of why, which is indistinguishable from a field that legitimately has no options.

</details>

*8 info-level observations excluded. Run with `--verbose` to include.*

*3 PR discussion threads considered.*

---
<details>
<summary>Completed in 4m 08s | $1.23</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 4m 08s wall · 9m 23s compute (Reviewers: 2m 36s · Synthesis: 1m 29s) |
| Cost | $1.23 |
| Tokens | 472.6k in / 35.5k out |
| Turns | 19 |

</details>
